pack.go

```diff
@@ -21,32 +21,61 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"time"
 
 	"github.com/opencontainers/go-digest"
+	artifactspec "github.com/oras-project/artifacts-spec/specs-go/v1"
+
 	specs "github.com/opencontainers/image-spec/specs-go"
 	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
 	"oras.land/oras-go/v2/content"
 	"oras.land/oras-go/v2/errdef"
 )
 
-// MediaTypeUnknownConfig is the default mediaType used when no
-// config media type is specified.
-const MediaTypeUnknownConfig = "application/vnd.unknown.config.v1+json"
+const (
+	// MediaTypeUnknownConfig is the default mediaType used when no
+	// config media type is specified.
+	MediaTypeUnknownConfig = "application/vnd.unknown.config.v1+json"
+
+	// AnnotationArtifactCreated is the annotation key representing the UTC date
+	// and time on which the artifact was created, in RFC3339 format.
+	// References:
+	// - https://datatracker.ietf.org/doc/html/rfc3339#section-5.6
+	// - https://github.com/oras-project/artifacts-spec/blob/main/artifact-manifest.md#oras-artifact-manifest-properties
+	AnnotationArtifactCreated = "io.cncf.oras.artifact.created"
+)
+
+// ErrMissingArtifactType is returned by PackArtifact() when no artifact type is
+// specified.
+var ErrMissingArtifactType = errors.New("missing artifact type")
 
 // PackOptions contains parameters for oras.Pack.
 type PackOptions struct {
-	ConfigDescriptor    *ocispec.Descriptor
-	ConfigMediaType     string
-	ConfigAnnotations   map[string]string
+	// ConfigDescriptor is a pointer to the descriptor of the config blob.
+	ConfigDescriptor *ocispec.Descriptor
+	// ConfigMediaType is the media type of the config blob.
+	// If not specified, MediaTypeUnknownConfig will be used.
+	ConfigMediaType string
+	// ConfigAnnotations is the annotation map of the config descriptor.
+	ConfigAnnotations map[string]string
+	// ManifestAnnotations is the annotation map of the manifest.
+	ManifestAnnotations map[string]string
+}
+
+// PackArtifactOptions contains parameters for oras.PackArtifact.
+type PackArtifactOptions struct {
+	// Subject is the subject of the ORAS Artifact Manifest.
+	Subject *artifactspec.Descriptor
+	// ManifestAnnotations is the annotation map of the manifest.
 	ManifestAnnotations map[string]string
 }
 
 // Pack packs the given layers, generates a manifest for the pack,
-// and pushes it to a content storage. If succeeded, returns a descriptor of the manifest.
+// and pushes it to a content storage.
+// If succeeded, returns a descriptor of the manifest.
 func Pack(ctx context.Context, pusher content.Pusher, layers []ocispec.Descriptor, opts PackOptions) (ocispec.Descriptor, error) {
```

Comment:
Should we push the layers for users? Otherwise we may want to call out "layers must be pushed before Pack() is called.".